ls1intum/Artemis
# `General`: Disable loading buttons

### Checklist
#### General


#### Client

### Motivation and Context


In some edge cases a button could be pressed multiple times which lead to some bugs in Artemis. E.g. pressing "Start exercise" for a programming exercise could lead to multiple submissions.

### Description

A button with a loading animation is now also automatically disabled.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Try to start the exercise multiple times by spamming "Start exercise" (an auto clicker might help here)
3. Check if the submissions are only registered once. This can be done in two ways:
- Go to the repository and check the settings on the left side. The "Weebhooks" tab should only contain one web hook
- Submit something as a student and check the amount of submissions in the participation overview of the exercise

### Review Progress





#### Code Review
#### Manual Tests
